### PR TITLE
408: roles for the members

### PR DESCRIPTION
**Pull request by Nesssu**
The admins can now update the roles and admin permissions for the members and also delete them from the Members page.

Also changed the CSS of the modals a little bit which causes issues on the way the confirm modals look when attenting to an event. But because they will be removed, I didn't waste time on fixing them.

Also these are on Trello as three separate tickets but I made them into a single PR to make it easier to test

### CODE REVIEW

**Comment by Nesssu** [comment]
[408 - Roles for the members](https://trello.com/c/X4DNS1Ud/104-408-roles-for-the-members)

**Comment by Nesssu** [comment]
[407 - Add/Remove permissions for users](https://trello.com/c/WWeVIThu/103-407-add-remove-permissions-for-users)

**Comment by Nesssu** [comment]
[406 - Delete accounts](https://trello.com/c/2a6XERw5/102-406-delete-accounts)

File: server/routes/api.js
Comment (Jezerig) [inline server/routes/api.js:1]: image

I think you could exclude the password just by doing this

**Review by Jezerig — COMMENTED** [review]
The members page looks really nice now! Only thing I would add is a way for the user to see some sort of highlight that shows themselves in the list of members. Also, the interaction when an admin deletes their own account that they're currently logged in on is weird and leaves them in a limbo.

**Comment by Nesssu** [comment]
Now the user can't delete their own account. When they open to manage modal, the delete button is disabled. We should probably allow users to leave the association from their profile page thought. Also the current user is highlighted in members list. And the way to not get the password is a lot better so changed it to that.

**Review by Jezerig — APPROVED** [review]

**Review by Enlarem — APPROVED** [review]

File: client/src/components/Member.js [hunk 1]
```diff
@@ -5,30 +5,114 @@ import CardContent from '@mui/material/CardContent';
...
+import ManageMemberModal from '../modals/ManageMemberModal';
 
-function Member({member}) {
+function Member(props) {
+  const [openManageMemberModal, setOpenManageMemberModal] = useState(false);
+
+  // Delete the memner from the association
+  const deleteMember = async () =>
+  {
+    fetch('/api/delete/member/' + props.member.id, {
```
Comment (jannejjj) [inline client/src/components/Member.js:18]: the param should be props.currentUser.id I guess, noticed this when trying to delete a member. Other than this, everything looks really good! OK to merge after fixing this. 🤝

**Pull request merged**